UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -238,32 +243,50 @@ private async Task<string> SelectLatestPublishedDateBatch(List<Products> product
+                throw new OperationCanceledException();
+            }
             if (CheckProductDoesExistInResponseItem(item, productItem) && productItem.Cancellation != null && productItem.Cancellation.UpdateNumber.HasValue
                                     && Convert.ToInt32(updateNumber) == productItem.Cancellation.UpdateNumber.Value)
             {
-                CheckProductWithCancellationData(internalSearchBatchResponse, productList, item, productItem, compareProducts);
+                await CheckProductWithCancellationData(internalSearchBatchResponse, productList, item, productItem, compareProducts, message, cancellationTokenSource, cancellationToken, exchangeSetRootPath);
             }
             else if (CheckProductDoesExistInResponseItem(item, productItem)
             && CheckEditionNumberDoesExistInResponseItem(item, productItem) && CheckUpdateNumberDoesExistInResponseItem(item, productItem))
             {
                 internalSearchBatchResponse.Entries.Add(item);
                 productList.Add(compareProducts);
+                await StartPreparingForDownload(item, productItem, exchangeSetRootPath, message, cancellationTokenSource, cancellationToken);
             }
         }
 
-        private void CheckProductWithCancellationData(SearchBatchResponse internalSearchBatchResponse, List<string> productList, BatchDetail item, Products productItem, string compareProducts)
+        private async Task CheckProductWithCancellationData(SearchBatchResponse internalSearchBatchResponse, List<string> productList, BatchDetail item, Products productItem, string compareProducts, SalesCatalogueServiceResponseQueueMessage message, CancellationTokenSource cancellationTokenSource, CancellationToken cancellationToken, string exchangeSetRootPath)
         {
             var matchEditionNumber = item.Attributes.Where(a => a.Key == "EditionNumber").ToList();
             if (matchEditionNumber.Any(a => a.Value == productItem.Cancellation.EditionNumber.Value.ToString()))
             {
                 matchEditionNumber.ForEach(c => c.Value = Convert.ToString(productItem.EditionNumber));
                 internalSearchBatchResponse.Entries.Add(item);
                 productList.Add(compareProducts);
+                await StartPreparingForDownload(item, productItem, exchangeSetRootPath, message, cancellationTokenSource, cancellationToken);
             }
         }
 
+        private async Task StartPreparingForDownload(BatchDetail item, Products productItem, string exchangeSetRootPath, SalesCatalogueServiceResponseQueueMessage message, CancellationTokenSource cancellationTokenSource, CancellationToken cancellationToken)
```

Comment:
Is this method named right?  Seems to be more of a Download than a Prepare.  Maybe `PerformDownload` or `PerformBatchFileDownload`?

Reply:
Updated method name. Thanks!